plugins/lookup/passwordstore.py

```diff
@@ -261,19 +261,21 @@ def check_pass(self):
                     if ':' in line:
                         name, value = line.split(':', 1)
                         self.passdict[name.strip()] = value.strip()
+            if os.path.isfile(os.path.join(self.paramvals['directory'], self.passname + ".gpg")):
+                # Only accept password as found, if there a .gpg file for it (might be a tree node otherwise)
+                password_found = True
         except (subprocess.CalledProcessError) as e:
-            if e.returncode != 0 and 'not in the password store' in e.output:
-                # if pass returns 1 and return string contains 'is not in the password store.'
-                # We need to determine if this is valid or Error.
-                if self.paramvals['missing'] == 'error':
-                    raise AnsibleError('passwordstore: passname {0} not found and missing=error is set'.format(self.passname))
-                else:
-                    if self.paramvals['missing'] == 'warn':
-                        display.warning('passwordstore: passname {0} not found'.format(self.passname))
-                    return False
-            else:
+            # 'not in password store' is the expected error if a password wasn't found
+            if 'not in the password store' not in e.output:
```

Comment:
To fully negate the original `if` expression:
```suggestion
            if e.returncode == 0 or 'not in the password store' not in e.output:
```

Reply:
It’s impossible for return code to be 0 here, !=0 is the precondition for this exception to be thrown, see: https://docs.python.org/3/library/subprocess.html#exceptions

> [check_call()](https://docs.python.org/3/library/subprocess.html#subprocess.check_call) and [check_output()](https://docs.python.org/3/library/subprocess.html#subprocess.check_output) will raise [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) if the called process returns a non-zero return code.

hence my comment above:

> This change made the logic more obvious (e.g., getting rid of the superfluous e.returncode != 0 while handling CalledProcessError).

That said, it seems like the function above which is described as “backhacked check_output with input for python 2.7” actually does strange things, basically breaking the described contract if there is some special output in the output from the process, without altering return code.

So the correct solution is to correct check_output2 to actually set a returncode before throwing an exception (which is easy to accomplish).

Reply:
Good news is that is probably makes no difference in this code anyway.

Bad news is, that the checks in `check_output2` will break on localized environments (this message is translated). So this is not stable. I'll add something to work around this, let's see if you'll find it sufficient.